### PR TITLE
zim: use venv

### PR DESCRIPTION
**Pull request by branchv**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

---

### CODE REVIEW

**Comment by branchvincent** [comment]
Python bindings for `pygobject` are being installed outside its prefix:

```
$ python3.12 -c 'import gi; print(gi.__file__)'
/home/linuxbrew/.linuxbrew/Cellar/python@3.12/3.12.2_1/lib/python3.12/site-packages/gi/__init__.py
```

**Comment by branchvincent** [comment]
Yea nvm this is actually from build isolation (thought it was the same issue as `libcap-ng` in #163561)

```console
$ python3.12 -c 'import gi, pathlib; print(pathlib.Path(gi.__file__).resolve())'
/home/linuxbrew/.linuxbrew/Cellar/pygobject3/3.48.2/lib/python3.12/site-packages/gi/__init__.py
```

**Comment by branchvincent** [comment]
Although, not sure why it only seems to be a problem on Linux?

**Comment by cho-m** [comment]
> Although, not sure why it only seems to be a problem on Linux?

Might not be related, but I also saw OS-specific behavior in https://github.com/Homebrew/homebrew-core/pull/168155#issuecomment-2040755396 where macOS was using older (`python@3.12` bundled ?) `wheel` while Linux was using newer `wheel`.

For some reason, build isolated build on macOS is able to access more already installed packages (e.g. `wheel` and `pygobject`).

**Comment by cho-m** [comment]
I wonder if the `sitecustomize` path modifications on macOS could impact build isolation
https://github.com/Homebrew/homebrew-core/blob/ef0a1b2f202d1d93840cc24a94d53e8650d8ca44/Formula/p/python%403.12.rb#L451-L454

File: Formula/z/zim.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ class Zim < Formula
   end
 
   depends_on "pkg-config" => :build
+  depends_on "python-setuptools" => :build
```
Comment (ZhongRuoyu) [inline Formula/z/zim.rb:15]: Can we get rid of `python-setuptools`?

Comment (branchvincent) [reply Formula/z/zim.rb]: done, also switched to a venv since that is a more typical installation

**Comment by branchvincent** [comment]
let's merge with bottles since this now uses a venv (changing the file layout)

**Review by ZhongRuoyu — APPROVED** [review]

File: Formula/z/zim.rb (hunk 1 above)
Comment (ZhongRuoyu) [reply Formula/z/zim.rb]: Thanks!

**Pull request merged**